Authenticator/Source/TokenFormViewController.swift

```diff
@@ -27,14 +27,54 @@ import SVProgressHUD
 
 class TokenFormViewController: UITableViewController {
     private var form: TokenForm?
-    private var viewModel: TableViewModel<Form> = EmptyTableViewModel()
+    private var viewModel: TableViewModel<Form> = EmptyTableViewModel() {
+        didSet {
+            guard oldValue.sections.count == viewModel.sections.count else {
+                // Automatic updates aren't implemented for changing number of sections
+                tableView.reloadData()
+                return
+            }
+            tableView.beginUpdates()
+            for sectionIndex in oldValue.sections.indices {
+                let oldSection = oldValue.sections[sectionIndex]
+                let newSection = viewModel.sections[sectionIndex]
+                let changes = diff(from: ArraySlice(oldSection.rows), to: ArraySlice(newSection.rows), comparator: {
+                    switch ($0, $1) {
+                    case (.TextFieldRow, .TextFieldRow): return true
+                    case (.TokenTypeRow, .TokenTypeRow): return true
+                    case (.DigitCountRow, .DigitCountRow): return true
+                    case (.AlgorithmRow, .AlgorithmRow): return true
+                    default: return false
+                    }
+                })
+                for change in changes {
+                    switch change {
+                    case .Insert(let rowIndex):
+                        let indexPaths = [NSIndexPath(forRow: rowIndex, inSection: sectionIndex)]
+                        tableView.insertRowsAtIndexPaths(indexPaths, withRowAnimation: .Automatic)
+                    case .Update(let rowIndex):
+                        let indexPath = NSIndexPath(forRow: rowIndex, inSection: sectionIndex)
+                        updateRowAtIndexPath(indexPath)
+                    case .Delete(let rowIndex):
+                        let indexPaths = [NSIndexPath(forRow: rowIndex, inSection: sectionIndex)]
+                        tableView.deleteRowsAtIndexPaths(indexPaths, withRowAnimation: .Automatic)
+                    case let .Move(fromRowIndex, toRowIndex):
+                        let fromIndexPath = NSIndexPath(forRow: fromRowIndex, inSection: sectionIndex)
```

Comment:
Length Violation: Line should be 100 characters or less: currently 102 characters